### ISSUE TITLE
unable to get the diffWords count , how many words added and how words many removed

### ISSUE THREAD

**Issue by Ganeshkaredla** [kpdecker/jsdiff#312]
Hi @kpdecker 

I want to get the difference between `oldText` and `newText` words difference how many words added and how many words removed ounts.

Here is the approach that I followed to get words difference count

`
function getDiffCount(oldText,newText){
     const diffList = Diff.diffWords(oldText, newText, {})

      let addedCount = 0
      let removedCount = 0

      diffList.forEach(part => {
         addedCount = part.added ? part.count + addedCount : addedCount
         removedCount = part.removed ? part.count + removedCount : removedCount
      })

      return {
         added: addedCount,
         removed: removedCount
      }
}
`
From this code, I was unable to get an exact count of words difference because the white spaces were also included

For example:
`This is a sample text`
   
if i enter this text there are 5 words added here but by using the above code it was considering the spaces between words and giving cont as 9 as there are 4 spaces in the example text
     
 Is this approach is correct to get the count difference for added and removed?
 Help me to fix this.
 
 Thanks in advance.

**Comment by ExplodingCabbage** [comment]
Yeah, I agree that this behaviour is wrong. I think we can make it so that the tokens `diffWords` splits a text into each consist of a word plus its trailing whitespace, rather than the word and whitespace being distinct tokens, and then we can use a custom `.equals` method to make tokens that differ only by whitespace compare equal. That'll give us better diffs and better counts.

**Comment by ExplodingCabbage** [comment]
This is kinda done, now; in the code on master (not yet released to npm), tokens in diffWords are either words or punctuation marks, not runs of whitespace.
